Solutions/DNS Domain Solution for Microsoft Sentinel/Analytic Rules/ExcessiveNXDOMAINDNSQueriesAnomalyBased.yaml

```diff
@@ -0,0 +1,130 @@
+  - connectorId: Corelight
+    dataTypes:
+      - Corelight
+  - connectorId: InfobloxNIOS
+    dataTypes:
+      - Syslog
+  - connectorId: NXLogDnsLogs
+    dataTypes:
+      - NXLog_DNS_Server_CL
+  - connectorId: DNS
+    dataTypes:
+      - DnsEvents
+  - connectorId: AIVectraStream
+    dataTypes:
+      - VectraStream
+  - connectorId: WindowsForwardedEvents
+    dataTypes:
+      - WindowsEvents
+  - connectorId: Zscaler
+    dataTypes:
+      - CommonSecurityLog
+  - connectorId: ISCBind
+    dataTypes:
+      - Syslog
+
+queryFrequency: 1d
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CommandAndControl
+relevantTechniques:
+  - T1568
+  - T1008
+query: |
+  let threshold = materialize (_GetWatchlist('DNS_Solution_Monitoring_Configuration')
+    | where wl_RuleName == 'Anomaly - Excessive NXDOMAIN DNS Queries'
+        and wl_Type == 'Detection'
+    | project toint(wl_AnomalyThreshold));
+  let min_t = ago(14d);
+  let max_t = now();
+  let dt = 1d;
+  let summarizationexist = materialize(
+    union isfuzzy=true 
+        (
+        DNS_Summarized_Logs_ip_CL
+        | where EventTime_t > ago(1d) 
+        | project v = int(2)
+        ),
+        (
+        print int(1) 
+        | project v = print_0
+        )
+    | summarize maxv = max(v)
+    | extend sumexist = (maxv > 1)
+    );
+  let allData = union isfuzzy=true
+        (
+        (datatable(exists: int, sumexist: bool)[1, false]
+        | join (summarizationexist) on sumexist)
+        | join (
+            _Im_Dns(responsecodename='NXDOMAIN', starttime=todatetime(min_t), endtime=max_t)
+            //| where TimeGenerated > min_t
+            | summarize Count=count() by SrcIpAddr, bin(TimeGenerated, 1h)
+            | extend EventTime = TimeGenerated, Count = toint(Count), exists=int(1)
+            )
+            on exists
+        | project-away exists, maxv, sum*
+        ),
+        (
+        DNS_Summarized_Logs_ip_CL
+        | where EventTime_t > min_t and EventResultDetails_s == 'NXDOMAIN'
+        | summarize Count=toint(sum(count__d)) by SrcIpAddr=SrcIpAddr_s, bin(EventTime=EventTime_t, 1h)
+        );
+  allData
+  | make-series EventCount=sum(Count) on EventTime from min_t to max_t step dt by SrcIpAddr
+  | extend (anomalies, score, baseline) = series_decompose_anomalies(EventCount, toscalar(threshold), -1, 'linefit')
+  | mv-expand anomalies, score, baseline, EventTime, EventCount
+  | extend
+    anomalies = toint(anomalies),
+    score = toint(score),
+    baseline = toint(baseline),
+    EventTime = todatetime(EventTime),
+    Total = tolong(EventCount)
+  | where EventTime >= ago(dt)
+  | where score >= toscalar(threshold)
+  | join kind=inner(_Im_Dns(responsecodename='NXDOMAIN', starttime=ago(dt), endtime=max_t)
+    | summarize DNSQueries = make_set(DnsQuery) by SrcIpAddr)
+    on SrcIpAddr
+
+entityMappings:
+  - entityType: DNS
+    fieldMappings:
+      - identifier: DomainName
+        columnName: DNSQueries
```

Comment:
DnsQueries is an array, can you verify if this would work in the Analytical rule entity mapping? The field is expecting a string.

Reply:
This was getting saved in entity as an array. Changed the mapping of array from entity to custom